int_scan.cpp

```diff
@@ -0,0 +1,133 @@
+#include "int_scan.hpp"
+
+enum Base {OCT=8, DEC=10, HEX=16};
+
+/*
+The function returns a member of the Base enumeration, which represents the base of the input string.
+If the input string starts with the character 0 (zero), then the input string represents an octal number.
+If the input string starts with the characters 0x or 0X (zero x/X), then the input string represents a hexadecimal
+number.
+If the input string does not have one of these prefixes, then the input string represents a decimal number.
+The function starts looking for the prefix from the index specified by the position pointer.
+The function also returns the position of the next character of the input to be read after the prefix (if it exists).
+*/
+static inline Base check_base(const char *input, int *position)
+{
+    *position=0;
+    if (input[*position]!='0')
+        return DEC;
+    else
```

Comment:
No need to `else` here because above you return. So, if you didn't, you can just continue and avoid nesting. Same in other places in the code.

Reply:
i used the if-else statements instead of using seperate if statements for the code to be more easily understandable. my goal was better readabilty of the code, since with the if-else statement the condition that corresponds to a particular return value was more evident (or at least that's what I thought). Changing these coding segments is easy, but what's your opinion for future reference?

Reply:
That makes sense but also that way, the code implies that you try to avoid something by using the `else`. Moreover, nesting becomes bad for readability as the code gets larger.